services/replicator/metadataReconciler.go

```diff
@@ -451,37 +474,179 @@ func (r *metadataReconciler) reconcileDestExtent(destUUID string, localExtents m
+	// If that happens, this call becomes necessary to move the extent to consumed state(extent won't
+	// be moved to consumed state if it's not sealed on store)
+	err := r.sealExtentInStore(destUUID, extentUUID)
+	if err != nil {
+		r.logger.WithFields(bark.Fields{
+			common.TagErr: err,
+			common.TagDst: common.FmtDst(destUUID),
+			common.TagExt: common.FmtExt(extentUUID),
+		}).Error(`Failed to seal extent in store`)
+	} else {
+		r.logger.WithFields(bark.Fields{
+			common.TagErr: err,
+			common.TagDst: common.FmtDst(destUUID),
+			common.TagExt: common.FmtExt(extentUUID),
+		}).Info(`Extent sealed in store`)
+	}
+}
+
+func (r *metadataReconciler) sealExtentInMetadata(destUUID string, extentUUID string) {
+	updateRequest := &metadata.UpdateExtentStatsRequest{
+		ExtentUUID:      common.StringPtr(extentUUID),
+		DestinationUUID: common.StringPtr(destUUID),
+		Status:          common.MetadataExtentStatusPtr(shared.ExtentStatus_SEALED),
+	}
+	_, err := r.mClient.UpdateExtentStats(nil, updateRequest)
+	if err != nil {
+		r.logger.WithFields(bark.Fields{
+			common.TagErr: err,
+			common.TagDst: common.FmtDst(destUUID),
+			common.TagExt: common.FmtExt(extentUUID),
+		}).Error(`Failed to seal extent in metadata`)
+	} else {
+		r.logger.WithFields(bark.Fields{
+			common.TagDst: common.FmtDst(destUUID),
+			common.TagExt: common.FmtExt(extentUUID),
+		}).Info(`Extent sealed in metadata`)
+	}
+}
+
+func (r *metadataReconciler) sealExtentInStore(destUUID string, extentUUID string) error {
+	readExtentRequest := &metadata.ReadExtentStatsRequest{
+		DestinationUUID: common.StringPtr(destUUID),
+		ExtentUUID:      common.StringPtr(extentUUID),
+	}
+
+	lclLg := r.logger.WithFields(bark.Fields{
+		common.TagDst: common.FmtDst(destUUID),
+		common.TagExt: common.FmtExt(extentUUID),
+	})
+
+	readExtentResult, err := r.mClient.ReadExtentStats(nil, readExtentRequest)
+	if err != nil {
+		lclLg.WithField(common.TagErr, err).Error(`Metadata call ReadExtentStats failed`)
+		return err
+	}
+
+	sealReq := &store.SealExtentRequest{
+		ExtentUUID: common.StringPtr(extentUUID),
+	}
+
+	var errorOccured bool
+	for _, store := range readExtentResult.GetExtentStats().GetExtent().GetStoreUUIDs() {
+		storeClient, _, err := r.replicator.clientFactory.GetThriftStoreClientUUID(store, destUUID)
+		if err != nil {
+			lclLg.WithFields(bark.Fields{
+				common.TagErr:  err,
+				common.TagStor: common.FmtStor(store),
+			}).Error(`Failed to get store client`)
+			errorOccured = true
+			continue
+		}
+
+		ctx, cancel := thrift.NewContext(storeCallTimeout)
+		defer cancel()
+		err = storeClient.SealExtent(ctx, sealReq)
+		if err != nil {
+			lclLg.WithFields(bark.Fields{
+				common.TagErr:  err,
+				common.TagStor: common.FmtStor(store),
+			}).Error(`Failed to seal extent on store`)
+			errorOccured = true
+			continue
+		}
+	}
+
+	if errorOccured {
+		return errors.New(`seal extent on store failed for at least one replica`)
```

Comment:
the next reconciliation run will try again (and will keeping trying until it's sealed). I'm planning to add more metrics/stats in next diff to monitor this.